apple/internal/ios_rules.bzl

```
@@ -1113,6 +1113,18 @@ def _ios_dynamic_framework_impl(ctx):
 
     providers = processor_result.providers
 
+    new_providers = []
+    for provider in providers:
+        new_providers.append(provider)
+        if type(provider) == "AppleDynamicFramework":
+            # Make the ObjC provider using the framework_files depset found
+            # in the AppleDynamicFramework provider. This is to make the
+            # ios_dynamic_framework usable as a dependency in swift_library
+            objc_provider = apple_common.new_objc_provider(
+                dynamic_framework_file = provider.framework_files
+            )
+            new_providers.append(objc_provider)
+
```

Can build up `additional_providers`, add them to `providers`, then use `providers` down on 1137 instead of `new_providers`:

```suggestion
    additional_providers = []
    for provider in providers:
        if type(provider) == "AppleDynamicFramework":
            # Make the ObjC provider using the framework_files depset found
            # in the AppleDynamicFramework provider. This is to make the
            # ios_dynamic_framework usable as a dependency in swift_library
            objc_provider = apple_common.new_objc_provider(
                dynamic_framework_file = provider.framework_files
            )
            additional_providers.append(objc_provider)
    providers.extend(additional_providers)
```

Let's do this in the three locations.

---

Yep! I like that better. I'll make the update today and ping you when it's done.